modules/calcite/src/main/java/org/apache/ignite/internal/processors/query/calcite/metadata/IgniteMdCollation.java

```diff
@@ -548,9 +548,9 @@ public int compare(List<RexLiteral> o1, List<RexLiteral> o2) {
             case FULL:
                 for (RelCollation collation : leftCollations) {
                     for (RelFieldCollation field : collation.getFieldCollations()) {
-                        if (!(RelFieldCollation.NullDirection.LAST == field.nullDirection)) {
+                        if (!(RelFieldCollation.NullDirection.LAST.nullComparison ==
```

Comment:
If any method `enumerable<NodeName>` is called by Ignite optimizer?
Should they are removed?

Reply:
Cleaned up this mess a bit